pandas/tests/test_index.py

```diff
@@ -3748,15 +3748,23 @@ def test_numeric_compat2(self):
         tm.assert_index_equal(Index(result.values), expected, exact=True)
 
         # __floordiv__
-        idx = RangeIndex(0, 1000, 2)
-        result = idx // 2
-        expected = idx._int64index // 2
-        tm.assert_index_equal(result, expected, exact=True)
-
-        idx = RangeIndex(0, 1000, 1)
-        result = idx // 2
-        expected = idx._int64index // 2
-        tm.assert_index_equal(result, expected, exact=True)
+        cases_exact = [(RangeIndex(0, 1000, 2), 2, RangeIndex(0, 500, 1)),
+                       (RangeIndex(-99, -201, -3), -3, RangeIndex(33, 67, 1)),
+                       (RangeIndex(0, 1000, 1), 2,
+                        RangeIndex(0, 1000, 1)._int64index // 2),
+                       (RangeIndex(0, 100, 1), 2.0,
+                        RangeIndex(0, 100, 1)._int64index // 2.0)]
+        for idx, div, expected in cases_exact:
+            tm.assert_index_equal(idx // div, expected, exact=True)
+
+        cases_equiv = [(RangeIndex(), 50, RangeIndex()),
+                       (RangeIndex(2, 4, 2), 3, RangeIndex(0, 1, 1)),
+                       (RangeIndex(-5, -10, -6), 4, RangeIndex(-2, -1, 1)),
+                       (RangeIndex(-100, -200, 3), 2, RangeIndex())]
+        for idx, div, expected in cases_equiv:
+            result = idx // div
+            tm.assert_index_equal(result, expected, exact=False)
```

Comment:
For `len(self) == 1` and `len(self) == 0` the equivalent classes are particularly large, and it doesn't matter which representatives to return.  e.g. you could always return `RangeIndex()` for `len(self) == 0` or leave `_start` unchanged.  And since when `len(self) == 1`, `step // div` could be 0, I couldn't treat all cases the same. The choice of `step` and `stop` is arbitrary so I had `exact=False`.